realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -416,85 +398,85 @@ public static AllTypes createOrUpdateUsingJsonObject(Realm realm, JSONObject jso
         if (obj == null) {
             if (json.has("columnString")) {
                 if (json.isNull("columnString")) {
-                    obj = realm.createObject(AllTypes.class, null);
+                    obj = (AllTypesRealmProxy) realm.createObject(AllTypes.class, null);
                 } else {
-                    obj = realm.createObject(AllTypes.class, json.getString("columnString"));
+                    obj = (AllTypesRealmProxy) realm.createObject(AllTypes.class, json.getString("columnString"));
                 }
             } else {
-                obj = realm.createObject(AllTypes.class);
+                obj = (AllTypesRealmProxy) realm.createObject(AllTypes.class);
             }
         }
         if (json.has("columnString")) {
             if (json.isNull("columnString")) {
-                obj.setColumnString(null);
+                ((AllTypesRealmProxyInterface) obj).realmSet$columnString(null);
```

Comment:
I thought about that but then accessing `realm` or `row` would not be possible. This casting should only be something for the compiler and should not result in any bytecode AFAIK